src/runtime/cli/magent/main.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"flag"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/utils"
+
+	"github.com/kata-containers/kata-containers/src/runtime/pkg/magent"
+	"github.com/sirupsen/logrus"
+)
+
+var metricListenAddr = flag.String("listen-address", ":8090", "The address to listen on for HTTP requests.")
+var containerdAddr = flag.String("containerd-address", "/run/containerd/containerd.sock", "Containerd address to accept client requests.")
+var containerdConfig = flag.String("containerd-conf", "/etc/containerd/config.toml", "Containerd config file.")
+var logLevel = flag.String("log-level", "info", "Log level of logrus(trace/debug/info/warn/error/fatal/panic).")
+var logPath = flag.String("log-path", "", "Path where logs save to. Default is stdio.")
+
+var logFile *os.File
+
+func main() {
+	flag.Parse()
+
+	// init logrus
+	initLog()
+
+	// close log file if needed
+	defer func() {
+		if logFile != nil {
+			logFile.Close()
+		}
+	}()
+
+	// create new MAgent
+	ma, err := magent.NewMAgent(*containerdAddr, *containerdConfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// setup handlers, now only metrics is supported
```

Comment:
For my own curiosity: you say only metrics for now -- are you looking at a seperate handler in the future for events? Or, what else?

Reply:
Yes, new handler entrypoint can be added here.